src/main/java/no/hvl/dat250/h2020/group5/service/VoteService.java

```diff
@@ -14,45 +14,49 @@
 import java.time.Instant;
 import java.util.Optional;
 
-
 @Service
 public class VoteService {
 
-    final
-    PollRepository pollRepository;
+    final PollRepository pollRepository;
 
-    final
-    VoterRepository voterRepository;
+    final VoterRepository voterRepository;
 
-    final
-    VoteRepository voteRepository;
+    final VoteRepository voteRepository;
 
     final UserRepository userRepository;
 
-    final
-    StringToAnswerType stringToAnswerType = new StringToAnswerType();
+    final StringToAnswerType stringToAnswerType = new StringToAnswerType();
 
-    public VoteService(PollRepository pollRepository, VoterRepository voterRepository, VoteRepository voteRepository,
-                       UserRepository userRepository) {
+    public VoteService(
+            PollRepository pollRepository,
+            VoterRepository voterRepository,
+            VoteRepository voteRepository,
+            UserRepository userRepository) {
         this.pollRepository = pollRepository;
         this.voterRepository = voterRepository;
         this.voteRepository = voteRepository;
         this.userRepository = userRepository;
     }
 
     public Vote vote(CastVoteRequest castVoteRequest) {
+        if (castVoteRequest.getPollId() == null
+                || castVoteRequest.getUserId() == null
+                || castVoteRequest.getVote() == null) {
+            return null;
+        }
+
         Optional<Poll> p = pollRepository.findById(castVoteRequest.getPollId());
         Optional<Voter> u = voterRepository.findById(castVoteRequest.getUserId());
         AnswerType answer = stringToAnswerType.convert(castVoteRequest.getVote());
 
-        if (p.isEmpty() || p.get().getStartTime() == null || u.isEmpty() || answer == null ){
+        if (p.isEmpty() || p.get().getStartTime() == null || u.isEmpty() || answer == null) {
             return null;
         }
 
-        //Checking if the vote is cast before poll ended.
+        // Checking if the vote is cast before poll ended.
         Instant startTime = p.get().getStartTime().toInstant();
         Instant startTimePlusDuration = startTime.plusSeconds(p.get().getPollDuration());
```

Comment:
Kunne vi trukket dette ut i en funksjon? 

```suggestion
        private boolean voteIsCastedBeforePollIsEnded() {
           // Checking if the vote is cast before poll ended.
           Instant startTime = p.get().getStartTime().toInstant();
           Instant startTimePlusDuration = startTime.plusSeconds(p.get().getPollDuration()); 
           return Instant.now().isAfter(startTimePlusDuration);
        }
        
        if (voteIsCastedBeforePollIsEnded()) {
        ...
        }
```

Reply:
godt poeng 👍